### PR TITLE
PR: Add installer and environment info to issue reporter

### LINKED ISSUE #18072: Test #15173 with added info
### What steps will reproduce the problem?

aldfkja;ldfj ;alsfj a;lkdf jas;dklf ds;klf a;dls fa;dlsf

## Versions

* Spyder version: 5.4.0.dev0 
* Installer: False
* Python version: 3.10.4
* Qt version: 5.15.2
* PyQt5 version: 5.15.4
* Operating System: Darwin 20.6.0

### Dependencies

```
# Mandatory:
applaunchservices >=0.1.7            :  0.3.0 (OK)
atomicwrites >=1.2.0                 :  1.4.0 (OK)
chardet >=2.0.0                      :  4.0.0 (OK)
cloudpickle >=0.5.0                  :  2.1.0 (OK)
cookiecutter >=1.6.0                 :  1.7.3 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree >=3.0.2                 :  3.0.2 (OK)
IPython >=7.31.1;<8.0.0              :  7.33.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.1 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.5.1 (OK)
keyring >=17.0.0                     :  23.5.1 (OK)
nbconvert >=4.0                      :  6.5.0 (OK)
numpydoc >=0.6.0                     :  1.3.1 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.1 (OK)
pygments >=2.0                       :  2.12.0 (OK)
pylint >=2.5.0                       :  2.13.9 (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0                 :  1.4.1 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.1.10                   :  0.2.1 (OK)
qtawesome >=1.0.2                    :  1.1.1 (OK)
qtconsole >=5.3.0;<5.4.0             :  5.4.0.dev0 (OK)
qtpy >=2.1.0                         :  2.1.0 (OK)
rtree >=0.9.7                        :  1.0.0 (OK)
setuptools >=49.6.0                  :  62.3.2 (OK)
sphinx >=0.6.6                       :  4.5.0 (OK)
spyder_kernels >=2.3.1;<2.4.0        :  2.4.0.dev0 (OK)
textdistance >=4.2.0                 :  4.2.2 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog >=0.10.3                    :  2.1.8 (OK)
zmq >=22.1.0                         :  23.0.0 (OK)

# Optional:
cython >=0.21                        :  0.29.30 (OK)
matplotlib >=3.0.0                   :  3.5.2 (OK)
numpy >=1.7                          :  1.22.4 (OK)
pandas >=1.1.1                       :  1.4.2 (OK)
scipy >=0.17.0                       :  1.8.1 (OK)
sympy >=0.7.3                        :  1.10.1 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)
```

### Environment

<details><summary>Environment</summary>

```
Package                       Version     Editable project location
----------------------------- ----------- ----------------------------------------------------------------------------
alabaster                     0.7.12
altgraph                      0.17.2
applaunchservices             0.3.0
appnope                       0.1.3
arrow                         1.2.2
astroid                       2.11.5
atomicwrites                  1.4.0
attrs                         21.4.0
autopep8                      1.6.0
Babel                         2.10.1
backcall                      0.2.0
bcrypt                        3.2.2
beautifulsoup4                4.11.1
binaryornot                   0.4.4
black                         22.3.0
bleach                        5.0.0
certifi                       2022.5.18.1
cffi                          1.15.0
chardet                       4.0.0
charset-normalizer            2.0.12
click                         8.1.3
cloudpickle                   2.1.0
coloredlogs                   15.0.1
cookiecutter                  1.7.3
cryptography                  37.0.2
cycler                        0.11.0
Cython                        0.29.30
debugpy                       1.6.0
decorator                     5.1.1
defusedxml                    0.7.1
diff-match-patch              20200713
dill                          0.3.5.1
dmgbuild                      1.5.2
docutils                      0.17.1
ds-store                      1.3.0
entrypoints                   0.4
fastjsonschema                2.15.3
flake8                        4.0.1
fonttools                     4.33.3
humanfriendly                 10.0
idna                          3.3
imagesize                     1.3.0
importlib-metadata            4.11.4
inflection                    0.5.1
intervaltree                  3.1.0
ipykernel                     6.13.0
ipython                       7.33.0
ipython-genutils              0.2.0
isort                         5.10.1
jedi                          0.18.1
jellyfish                     0.9.0
Jinja2                        3.1.2
jinja2-time                   0.2.0
jsonschema                    4.5.1
jupyter-client                7.3.1
jupyter-core                  4.10.0
jupyterlab-pygments           0.2.2
keyring                       23.5.1
kiwisolver                    1.4.2
lazy-object-proxy             1.7.1
mac-alias                     2.2.0
macholib                      1.16
MarkupSafe                    2.1.1
matplotlib                    3.5.2
matplotlib-inline             0.1.3
mccabe                        0.6.1
mistune                       0.8.4
modulegraph                   0.19.2
mpmath                        1.2.1
mypy-extensions               0.4.3
nbclient                      0.6.3
nbconvert                     6.5.0
nbformat                      5.4.0
nest-asyncio                  1.5.5
numpy                         1.22.4
numpydoc                      1.3.1
packaging                     21.3
pandas                        1.4.2
pandocfilters                 1.5.0
paramiko                      2.11.0
parso                         0.8.3
pathspec                      0.9.0
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.1.1
pip                           22.1.1
platformdirs                  2.5.2
pluggy                        1.0.0
poyo                          0.5.0
prompt-toolkit                3.0.29
psutil                        5.9.1
ptyprocess                    0.7.0
py2app                        0.28.2
pycodestyle                   2.8.0
pycparser                     2.21
pydocstyle                    6.1.1
pyflakes                      2.4.0
Pygments                      2.12.0
pylint                        2.13.9
pyls-spyder                   0.4.0
PyNaCl                        1.5.0
pyobjc-core                   8.5
pyobjc-framework-Cocoa        8.5
pyobjc-framework-CoreServices 8.5
pyobjc-framework-FSEvents     8.5
pyparsing                     3.0.9
PyQt5                         5.15.6
PyQt5-Qt5                     5.15.2
PyQt5-sip                     12.10.1
PyQtWebEngine                 5.15.5
PyQtWebEngine-Qt5             5.15.2
pyrsistent                    0.18.1
python-dateutil               2.8.2
python-lsp-black              1.2.1
python-lsp-jsonrpc            1.0.0
python-slugify                6.1.2
pytz                          2022.1
pyxdg                         0.27
pyzmq                         23.0.0
QDarkStyle                    3.0.3       /Users/rclary/Documents/Repos/Spyder-IDE/spyder/external-deps/qdarkstyle
qstylizer                     0.2.1
QtAwesome                     1.1.1
qtconsole                     5.4.0.dev0  /Users/rclary/Documents/Repos/Spyder-IDE/spyder/external-deps/qtconsole
QtPy                          2.1.0
requests                      2.27.1
rope                          1.1.1
Rtree                         1.0.0
scipy                         1.8.1
setuptools                    62.3.2
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.3.2.post1
Sphinx                        4.5.0
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.4.0.dev0  /Users/rclary/Documents/Repos/Spyder-IDE/spyder
spyder-kernels                2.4.0.dev0  /Users/rclary/Documents/Repos/Spyder-IDE/spyder/external-deps/spyder-kernels
spyder-terminal               1.2.2
sympy                         1.10.1
terminado                     0.15.0
text-unidecode                1.3
textdistance                  4.2.2
three-merge                   0.1.1
tinycss2                      1.1.1
toml                          0.10.2
tomli                         2.0.1
tornado                       6.1
traitlets                     5.2.1.post0
ujson                         5.3.0
urllib3                       1.26.9
watchdog                      2.1.8
wcwidth                       0.2.5
webencodings                  0.5.1
wheel                         0.37.1
wrapt                         1.14.1
wurlitzer                     3.0.2
yapf                          0.32.0
zipp                          3.8.0

```
</details>

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

The Spyder version text in the About dialogue and the issue reporter now includes the Spyder installer type (standalone, conda, or pip).

Additionally, the IPython console environment is optionally included (in detail view) when submitting issues to GitHub from the issue reporter.
If included, the environment list will be the result of `python -m pip list` if it is a non-conda environment and the result of `conda list --prefix <env-path>` if it is a conda environment.

The versions text for "Copy version info" in the About dialogue and in the issue reporter are consolidated to one function.

### About Dialogue
Screen Shot 2022-05-31 at 9 58 33 AMScreen Shot 2022-05-31 at 11 12 09 AM

Example "Copy version info"

* Spyder version: 5.4.0.dev0 744e2ab2b (conda)
* Python version: 3.10.4 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.4
* Operating System: Darwin 20.6.0

### Issue Reporter Dialogue
Screen Shot 2022-05-31 at 1 08 19 PM
<details><summary>Example GitHub issue text</summary>

## Versions

* Spyder version: 5.4.0.dev0 744e2ab2b (conda)
* Python version: 3.10.4 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.4
* Operating System: Darwin 20.6.0

### Dependencies

```
# Mandatory:
applaunchservices >=0.1.7     :  0.3.0 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.1.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.31.1;<8.0.0       :  7.33.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.5.1 (OK)
keyring >=17.0.0              :  23.5.1 (OK)
nbconvert >=4.0               :  6.5.0 (OK)
numpydoc >=0.6.0              :  1.3.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.1 (OK)
pygments >=2.0                :  2.12.0 (OK)
pylint >=2.5.0                :  2.13.9 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0          :  1.4.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.3.0;<5.4.0      :  5.4.0.dev0 (OK)
qtpy >=2.1.0                  :  2.1.0 (OK)
rtree >=0.9.7                 :  1.0.0 (OK)
setuptools >=49.6.0           :  62.3.2 (OK)
sphinx >=0.6.6                :  4.5.0 (OK)
spyder_kernels >=2.3.1;<2.4.0 :  2.4.0.dev0 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.8 (OK)
zmq >=22.1.0                  :  23.0.0 (OK)

# Optional:
cython >=0.21                 :  0.29.30 (OK)
matplotlib >=3.0.0            :  3.5.2 (OK)
numpy >=1.7                   :  1.22.4 (OK)
pandas >=1.1.1                :  1.4.2 (OK)
scipy >=0.17.0                :  1.8.1 (OK)
sympy >=0.7.3                 :  1.10.1 (OK)
```
</details>

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @mrclary

### CODE REVIEW

**Comment by mrclary** [comment]
@spyder-ide/core-developers, what do you think of this PR?

**Comment by andfoy** [comment]
I find it great to trace if the issue reports are only restricted to the installers

**Comment by ccordoba12** [comment]
> The issue text sent to GitHub from the issue reporter now includes whether Spyder is launched from an installer or not.

I think this is a great idea! It'd be great too if we could also report when Spyder is installed under Anaconda.

> Additionally, the Python environment is listed (in detail view) except when Spyder is launched from an installer and the default interpreter is used.

I don't think this is a good idea because users could have their own internal and closed source packages installed, whose names and versions they most probably wouldn't like to see exposed publicly (at least I wouldn't like that).

Furthermore, we rarely need the full environment information to diagnose a problem, just the Spyder deps one.

**Comment by mrclary** [comment]
> > The issue text sent to GitHub from the issue reporter now includes whether Spyder is launched from an installer or not.
> 
> I think this is a great idea! It'd be great too if we could also report when Spyder is installed under Anaconda.

Okay, maybe instead of the extra line "Installer: True|False", I could append the Spyder version "Spyder version: <version> <revision> (installer|conda)"

> > Additionally, the Python environment is listed (in detail view) except when Spyder is launched from an installer and the default interpreter is used.
> 
> I don't think this is a good idea because users could have their own internal and closed source packages installed, whose names and versions they most probably wouldn't like to see exposed publicly (at least I wouldn't like that).
> 
> Furthermore, we rarely need the full environment information to diagnose a problem, just the Spyder deps one.

Okay, I understand. In my limited experience, I've had to ask reporters to post that info, but I certainly understand your concerns.

Is it worth considering having a checkbox (default deselected) that allows the user to supply that info if they choose? I think it would be convenient, particularly for less experienced users that launch from a conda environment (which they may have messed up). Just a thought.

**Comment by mrclary** [comment]
Also, perhaps I should add the "installer|conda" info to the version info of the "About Spyder" window, so that when users "Copy version info" it is included and they can paste into GitHub issues.

**Comment by ccordoba12** [comment]
> Okay, maybe instead of the extra line "Installer: True|False", I could append the Spyder version "Spyder version: <version> <revision> (installer|conda)"

Yep, I like that.

> Is it worth considering having a checkbox (default deselected) that allows the user to supply that info if they choose?

Great idea! For these things I much prefer opt-in than opt-out.

> Also, perhaps I should add the "installer|conda" info to the version info of the "About Spyder" window, so that when users "Copy version info" it is included and they can paste into GitHub issues.

I saw that you did it in the screenshots above and I find it a really nice touch.

**Comment by ccordoba12** [comment]
Also, about the hanging tests on Linux, I think they are not related to the work you're doing here. We're discussing with @dalthviz what to do about that.

**Comment by mrclary** [comment]
> Also, about the hanging tests on Linux, I think they are not related to the work you're doing here. We're discussing with @dalthviz what to do about that.

`test_completions_environment` is also failing on the Mac tests. I see that it is supposed to be skipped if not a conda environment, which is strange since we are only testing "pip" environments right now. However, the test environment looks to be a conda environment but with mostly pip installed packages. Perhaps I don't completely understand what is supposed to happen there, but it seems to me that we should be making a venv virtual environment for the pip tests. `is_anaconda` will return True since the executable is in a conda environment, and the test won't be skipped.

**Comment by CAM-Gerlach** [comment]
FWIW, its very useful to have the install type, since very often that's been a significant factor in errors in the past, and I've had to deduce that implicitly from checking if the versions matched the ones we'd pinned for our installers.

File: spyder/__init__.py [hunk 1]
```diff
@@ -75,27 +61,52 @@ def get_versions(reporev=True):
...
                        else qtpy.PYQT_VERSION),
-        'system': system,   # Linux, Windows, ...
+        'system': platform.system(),   # Linux, Windows, ...
         'release': platform.release(),  # XP, 10.6, 2.2.0, etc.
         'revision': revision,  # '9fdf926eccce',
         'branch': branch,  # '4.x' or master
     }
+
+
+def get_versions_text(reporev=True):
```
Comment (ccordoba12) [inline spyder/__init__.py:100]: This function is missing a docstring.

File: spyder/widgets/about.py [hunk 2]
```diff
@@ -14,15 +14,18 @@
...
+                            QScrollArea, QTabWidget)
 
 # Local imports
-from spyder import (__project_url__, __forum_url__,
-                    __trouble_url__, __website_url__, get_versions)
+from spyder import (__project_url__ as project_url,
+                    __forum_url__ as forum_url,
+                    __trouble_url__ as trouble_url,
+                    __website_url__ as website_url,
+                    get_versions, get_versions_text)
```
Comment (ccordoba12) [inline spyder/widgets/about.py:24]:
```suggestion
from spyder import (
    __project_url__ as project_url,
    __forum_url__ as forum_url,
    __trouble_url__ as trouble_url,
    __website_url__ as website_url,
    get_versions, get_versions_text
)
```

Style improvement.

File: spyder/widgets/about.py [hunk 3]
```diff
@@ -120,25 +113,19 @@ def __init__(self, parent):
...
-                github_url=__project_url__,
-                trouble_url=__trouble_url__,
-                forum_url=__forum_url__,
-                font_family=font_family,
-                font_size=font_size,
-            ))
-        self.label_legal = QLabel((
-            """
+            </div>""")
+        self.label_legal = QLabel(
```
Comment (ccordoba12) [inline spyder/widgets/about.py:117]:
```suggestion

        self.label_legal = QLabel(
```

For readability.

File: spyder/widgets/reporterror.py [hunk 4]
```diff
@@ -21,11 +21,14 @@
 
 # Local imports
 from spyder import (__project_url__, __trouble_url__, dependencies,
-                    get_versions)
-from spyder.config.base import _, running_under_pytest
+                    get_versions_text)
+from spyder.config.base import _, is_pynsist, running_in_mac_app
 from spyder.config.gui import get_font
+from spyder.config.manager import CONF
```
Comment (ccordoba12) [inline spyder/widgets/reporterror.py:27]: Please remove `CONF` from here. Instead, you need to inherit the `SpyderConfigurationAccessor` mixin in `SpyderErrorDialog` and then use its `get_conf` method to get configuration options.

File: spyder/widgets/reporterror.py [hunk 5]
```diff
@@ -246,8 +249,13 @@ def __init__(self, parent=None, is_report=False):
...
         layout.addSpacing(15)
-        layout.addWidget(self.dismiss_box)
-        layout.addSpacing(15)
+        if not self.is_report:
+            layout.addWidget(self.dismiss_box)
+            layout.addSpacing(15)
+        if (not (is_pynsist() or running_in_mac_app())
+                or not CONF.get('main_interpreter', 'default')):
+            layout.addWidget(self.include_env)
+            layout.addSpacing(15)
```
Comment (ccordoba12) [inline spyder/widgets/reporterror.py:258]:
```suggestion

        if not self.is_report:
            layout.addWidget(self.dismiss_box)
            layout.addSpacing(15)

        if (
            not (is_pynsist() or running_in_mac_app())
            or not CONF.get('main_interpreter', 'default')
        ):
            layout.addWidget(self.include_env)
            layout.addSpacing(15)

```

For readability.

**Review by ccordoba12 — COMMENTED** [review]
This is a really nice improvement, thanks @mrclary!

**Comment by ccordoba12** [comment]
`spyder/widgets/tests/test_reporterror.py::test_render_issue` is failing on Windows now.

**Comment by mrclary** [comment]
Okay, changing render_issue to class method should fix it.

File: spyder/widgets/reporterror.py [hunk 6]
```diff
@@ -246,8 +249,18 @@ def __init__(self, parent=None, is_report=False):
         layout.addWidget(self.details)
         layout.addWidget(self.desc_chars_label)
         layout.addSpacing(15)
-        layout.addWidget(self.dismiss_box)
-        layout.addSpacing(15)
+
+        if not self.is_report:
+            layout.addWidget(self.dismiss_box)
+            layout.addSpacing(15)
```
Comment (ccordoba12) [inline spyder/widgets/reporterror.py:255]:
```suggestion
```

There's no need of this because it introduces unnecessary spacing between the checkboxes.

**Review by ccordoba12 — APPROVED** [review]
Thanks @mrclary!

**Pull request merged**